anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -483,201 +482,240 @@ def image(dbsession, request_inputs, bodycontent=None):
-            except AnchoreApiError:
-                raise
-            except Exception as err:
-                logger.exception("Error adding image")
-                httpcode = 404
-                raise err
+    return return_object, httpcode
 
-            if image_record:
-                httpcode = 200
-                return_object = image_record
-            else:
-                httpcode = 404
-                raise Exception("could not add input image")
 
-    except AnchoreApiError as err:
-        logger.exception("Error processing image request")
-        return_object = anchore_engine.common.helpers.make_response_error(
-            err.message, in_httpcode=err.__response_code__, details=err.detail
+def get_dockerfile_info(jsondata):
+    dockerfile = None
+    dockerfile_mode = None
+    if "dockerfile" in jsondata:
+        dockerfile = jsondata["dockerfile"]
+        try:
+            # this is a check to ensure the input is b64 encoded
+            base64.decodebytes(dockerfile.encode("utf-8"))
+            dockerfile_mode = "Actual"
+        except Exception as err:
+            raise Exception(
+                "input dockerfile data must be base64 encoded - exception on decode: "
+                + str(err)
+            )
+    return dockerfile, dockerfile_mode
+
+
+def resolve_image_info_overrides_and_input_string(params, image_info, input_string):
+    image_info_overrides = {}
+
+    input_tag = params.get("tag", None)
+    input_digest = params.get("digest", None)
+    if input_tag and input_digest:
+        input_fulldigest = "{}/{}@{}".format(
+            image_info["registry"], image_info["repo"], input_digest
+        )
+        image_info_overrides["fulltag"] = input_tag
+        image_info_overrides["tag"] = image_info["tag"]
+        if params.get("created_at", None):
+            image_info_overrides["created_at_override"] = params.get("created_at")
+        input_string = input_fulldigest
+    return image_info_overrides, input_string
+
+
+def resolve_final_image_info(
+    account_id, input_string, registry_creds, dbsession, image_info_overrides
+):
+    try:
+        image_info = anchore_engine.common.images.get_image_info(
+            account_id,
+            "docker",
+            input_string,
+            registry_lookup=True,
+            registry_creds=registry_creds,
         )
-        httpcode = err.__response_code__
     except Exception as err:
-        logger.exception("Error processing image request")
-        return_object = anchore_engine.common.helpers.make_response_error(
-            err, in_httpcode=httpcode
+        fail_event = anchore_engine.subsys.events.ImageRegistryLookupFailed(
+            user_id=account_id,
+            image_pull_string=input_string,
+            data=err.__dict__,
+        )
+        try:
+            add_event(fail_event, dbsession)
+        except Exception:
+            logger.warn("Ignoring error creating image registry lookup event")
+        raise err
```

Comment:
similar to other broad exception comments

Reply:
This one looks like it's squelching an error when a "fail" event _fails_ to be submitted....and might need some thinking about re-implementation, because removing this WOULD change the functionality here (main goal was to refactor and modularize)